DuckDuckGo/YoutubePlayer/DuckPlayerOverlayPixels.swift

```diff
@@ -16,84 +16,93 @@
 //  limitations under the License.
 //
 import PixelKit
+import WebKit
 
 protocol DuckPlayerOverlayPixelFiring {
 
     var pixelFiring: PixelFiring? { get set }
-    var navigationHistory: [URL] { get set }
-
-    func handleNavigationAndFirePixels(url: URL?, duckPlayerMode: DuckPlayerMode)
+    var webView: WKWebView? { get set }
+    var duckPlayerMode: DuckPlayerMode { get set }
+    func fireNavigationPixelsIfNeeded(webView: WKWebView)
+    func fireReloadPixelIfNeeded(url: URL)
 }
 
-final class DuckPlayerOverlayUsagePixels: DuckPlayerOverlayPixelFiring {
+final class DuckPlayerOverlayUsagePixels: NSObject, DuckPlayerOverlayPixelFiring {
 
     var pixelFiring: PixelFiring?
-    var navigationHistory: [URL] = []
+    var duckPlayerMode: DuckPlayerMode = .disabled
+
+    weak var webView: WKWebView? {
+        didSet {
+            if let webView = webView {
```

Comment:
I understand that this setter is only called once, but to future-proof it you could call `removeObservers()` at the top.

Reply:
@ayoy here's the change for this.  Added a local flag, as you can't check if an observer is present, and removing a not-present observer will trigger a FAULT: NSRangeException

Reply:
looks good, thanks @afterxleep!